Nos/Service/NamesAPI.swift

```diff
@@ -51,15 +51,33 @@ class NamesAPI {
         throw Error.unexpected
     }
 
+    /// Verifies that a given username is free to claim in nos.social
     func verify(username: String, keyPair: KeyPair) async throws -> Bool {
+        try await verify(
+            username: username,
+            host: verificationURL,
+            keyPair: keyPair,
+            valueWhenNotFound: true
+        )
+    }
+
+    /// Verifies that a given username_at_host NIP-05 can be connected to the keyPair
+    ///
+    /// - Parameter valueWhenNotFound: What to return if the server returns 404
```

Comment:
What does it mean semantically when a 404 is returned? That the name is not already registered?

Reply:
I have trouble with semantics here. When verifying nos.social usernames, we know that a username is free to take if the server returns a 404 or if it was already associated to the npub in the past. When verifying external nip-05, we are not looking for free to take usernames, we just want to check if it is already associated.

While they are semantically, the code its almost the same. I'll think a different approach.

Reply:
I split verify into verify and checkAvailability. So, verify checks that a NIP-05 is recognized by the server while checkAvailability looks if a given username is free to take (server responded with 404) or was already recognizing that username for the npub is trying to check the availability.

Reply:
The new APIs are much easier for me to read, thanks!